### ISSUE TITLE
 Since updating to 4.0.6 I’ve lost the Disk Activity chart

### ISSUE THREAD

**Issue by sis-chris** [rockstor/rockstor-core#2297]
Hi, I wasn't getting much traction in the forum so I thought I'd add this as an issue.

As per the title since updating to 4.0.6 I've lost the Disk Activity chart in the dashboard. All the other "tiles" appear to be working ok. openSUSE Leap has all of the latest updates.

Below is the bad network request I located in Firefox as requested by forum user Marenz.

Keep up the good work
Chris

`{

  {
    "pageref": "page_1",
    "startedDateTime": "2021-03-22T13:24:45.654+13:00",
    "request": {
      "bodySize": 0,
      "method": "GET",
      "url": "https://10.24.x.x/socket.io/?EIO=3&transport=polling&t=NXNKTjK&sid=d7c88176be7848ed8ae0132a76b1a45d",
      "httpVersion": "HTTP/1.1",
      "headers": [
        {
          "name": "Host",
          "value": "10.24.x.x"
        },
        {
          "name": "User-Agent",
          "value": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:86.0) Gecko/20100101 Firefox/86.0"
        },
        {
          "name": "Accept",
          "value": "*/*"
        },
        {
          "name": "Accept-Language",
          "value": "en-GB,en;q=0.5"
        },
        {
          "name": "Accept-Encoding",
          "value": "gzip, deflate, br"
        },
        {
          "name": "Connection",
          "value": "keep-alive"
        },
        {
          "name": "Referer",
          "value": "https://10.24.x.x/home?cb-table_length=15"
        },
        {
          "name": "Cookie",
          "value": "io=78f09cba5f7b403781628dc85c60b3a2; io=6aff95eba56c4ee5855dd97e0b8c6741; csrftoken=P0qnDAIjYLb5Kx7GQYFljNkjrcYNJCCP; redirect=1; testing=1; sid=daf4de89994bd2469a3085a920c9bcd0; sessionid=6i9nn0xog9bvdpu2ekthkgv3nrgww455"
        }
      ],
      "cookies": [
        {
          "name": "io",
          "value": "78f09cba5f7b403781628dc85c60b3a2"
        },
        {
          "name": "io",
          "value": "6aff95eba56c4ee5855dd97e0b8c6741"
        },
        {
          "name": "csrftoken",
          "value": "P0qnDAIjYLb5Kx7GQYFljNkjrcYNJCCP"
        },
        {
          "name": "redirect",
          "value": "1"
        },
        {
          "name": "testing",
          "value": "1"
        },
        {
          "name": "sid",
          "value": "daf4de89994bd2469a3085a920c9bcd0"
        },
        {
          "name": "sessionid",
          "value": "6i9nn0xog9bvdpu2ekthkgv3nrgww455"
        }
      ],
      "queryString": [
        {
          "name": "EIO",
          "value": "3"
        },
        {
          "name": "transport",
          "value": "polling"
        },
        {
          "name": "t",
          "value": "NXNKTjK"
        },
        {
          "name": "sid",
          "value": "d7c88176be7848ed8ae0132a76b1a45d"
        }
      ],
      "headersSize": 603
    },
    "response": {
      "status": 400,
      "statusText": "BAD REQUEST",
      "httpVersion": "HTTP/1.1",
      "headers": [
        {
          "name": "Server",
          "value": "nginx/1.16.1"
        },
        {
          "name": "Date",
          "value": "Mon, 22 Mar 2021 00:25:46 GMT"
        },
        {
          "name": "Content-Type",
          "value": "text/plain"
        },
        {
          "name": "Content-Length",
          "value": "11"
        },
        {
          "name": "Connection",
          "value": "keep-alive"
        },
        {
          "name": "Keep-Alive",
          "value": "timeout=20"
        },
        {
          "name": "Access-Control-Allow-Origin",
          "value": "*"
        },
        {
          "name": "Access-Control-Allow-Credentials",
          "value": "true"
        }
      ],
      "cookies": [],
      "content": {
        "mimeType": "text/plain",
        "size": 11,
        "text": "Bad Request"
      },
      "redirectURL": "",
      "headersSize": 253,
      "bodySize": 264
    },
    "cache": {},
    "timings": {
      "blocked": 0,
      "dns": 0,
      "connect": 0,
      "ssl": 0,
      "send": 0,
      "wait": 60005,
      "receive": 0
    },
    "time": 60005,
    "_securityState": "secure",
    "serverIPAddress": "10.24.x.x",
    "connection": "443"
  }
]`

![1c782cdfc928a542a6dcb21e2303fade548d1239](https://user-images.githubusercontent.com/82616762/114942834-3b7a7c80-9e99-11eb-931f-ebfddbf5c7db.png)

**Comment by phillxnet** [comment]
@sis-chris 

From:
https://github.com/rockstor/rockstor-core/blob/master/src/rockstor/smart_manager/data_collector.py#L571-L578

We source this data from the following command:
```
cat /proc/diskstats
```
Could you paste the output of this command (between code quotes) to check on this end of things. As there is little that has changed in this respect code wise.

**Comment by phillxnet** [comment]
Linking to the subpost within the forum:
https://forum.rockstor.com/t/built-on-opensuse-testing-channel-live-early-adopters-developers-only/6592/70

**Comment by phillxnet** [comment]
@sis-chris I think a new forum thread may have been the way to go on the forum, as you intimated in the linked post. But no worries.

And thanks for the report there and here.

**Comment by sis-chris** [comment]
Hi Philip, I've resolved the issue. I remembered I had been trying to get Certbot working around the same time as the 4.0.6 update. All my reading pointed to having to install snappy/snapd to get it to work which I'd done. Having a look at Yum it showed me the repository for snappy no longer existed so I removed it and snappy related packages. Yum was also showing me some package conflicts even after cleaning/checking. I did a check using Zypper and all looked good so I installed Certbot, which included a number of python2 packages, from the production repository and as well as available updates and short story long the Disk Activity chart is ok again now.

Thanks
Chris
